src/mixins/object_interactivity.mixin.js

```diff
@@ -108,73 +108,63 @@
     },
 
     /**
-     * Draws borders of an object's bounding box.
-     * Requires public properties: width, height
-     * Requires public options: padding, borderColor
-     * @param {CanvasRenderingContext2D} ctx Context to draw on
-     * @param {Object} styleOverride object to override the object style
-     * @return {fabric.Object} thisArg
-     * @chainable
+     * @public override if necessary
```

Comment:
we could say that this function is here exactly for the scope of overriding. Otherwise someone could argue, why having a method on a class that does not need `this` at all.
I would just say override this function in order to draw a different control box. example: rounded corners, different border style.
We should also inform whihc is the state of the context in that situation to give the dev some extra information.
(rotated aligned with object base, centered on the center of the object, scaled to absorb object scale and container scale and zoom )

Reply:
The ctx is in the same state as in the object's render method right?

Reply:
Or maybe not. I am confused

Reply:
the ctx is not in the same state, because when you scale the object the controls do not grow.
i m also wrong the ctx is only centered and rotated, and is not scaled at all.

I m pointing out that `override in case of necessity` can be writte on top of any function or none.
Since you made this refactor with a keen eye on having it overridable for some reason, just indicate the reason as an example for future reader

Reply:
If I recall correctly it is rotated and translated to center. But not scaled or skewed...

Reply:
36b7608e